gatling-core/src/main/scala/io/gatling/core/check/ConditionalCheck.scala

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright 2011-2016 GatlingCorp (http://gatling.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+package io.gatling.core.check
+
+import java.util.UUID
+import com.typesafe.scalalogging.StrictLogging
+import io.gatling.commons.validation._
+import io.gatling.core.session.{ Expression, Session }
+
+object ConditionalCheck {
+  
+  implicit def conditionalCheckBuilder2Check[R, C <: Check[R]](conditionalCheckBuilder: ConditionalCheckBuilder[R, C]): ConditionalCheck[R, C] = conditionalCheckBuilder.build
+
+  def checkIf[R, C <: Check[R]](condition: Expression[Boolean])(thenCheck: C)(implicit cw: ConditionalCheckWrapper[R, C]): C = cw.wrap(ConditionalCheckBuilder((r: R, s: Session) => condition(s), thenCheck))
+
+  def checkIf[R, C <: Check[R]](condition: (R, Session) => Validation[Boolean])(thenCheck: C)(implicit cw: ConditionalCheckWrapper[R, C]): C = 
+    cw.wrap(ConditionalCheckBuilder((r: R,s: Session) => condition(r, s), thenCheck))
+

```

Comment:
This should go in CheckSupport so it's automatically available without requiring additional import.